src/components/MozEngineWarning.ts

```diff
@@ -0,0 +1,85 @@
+import { LitElement, html, css } from 'lit'
+
+export class MozEngineWarning extends LitElement {
+  isVisible: boolean = false
+  isLocalAiEnabled: boolean = false
+
+  static properties = {
+    isVisible: { type: Boolean },
+    isLocalAiEnabled: { type: Boolean },
+  }
+
+  constructor() {
+    super()
+  }
+
+  firstUpdated() {
+    this.initLocalStorageData()
+  }
+
+  async initLocalStorageData() {
+    this.isLocalAiEnabled = await browser.permissions.contains({
+      permissions: ['trialML'],
+    })
+    this.isVisible = !this.isLocalAiEnabled
+  }
+
+  handleEnable() {
+    ;(browser.permissions.request as any)({ permissions: ['trialML'] }).then(
```

Comment:
uh maybe i'm not familiar enough with typescript, but this leading semicolon is to avoid someone adding some other statement before?

maybe we could convert this to async / await